Wikipedia/Code/ArticleViewController+LinkPreviewing.swift

```diff
@@ -146,3 +146,43 @@ extension ArticleViewController {
         commitPreview(of: previewingViewController)
     }
 }
+
+// MARK: Peek/Pop for Lead Image of ArticleVC (iOS 13 and later - no equivalent functionality on iOS 12 and prior)
+@available(iOS 13.0, *)
+extension ArticleViewController: UIContextMenuInteractionDelegate {
+    func contextMenuInteraction(_ interaction: UIContextMenuInteraction, configurationForMenuAtLocation location: CGPoint) -> UIContextMenuConfiguration? {
+        // If gallery has not been opened on that article, self.mediaList is nil - and we need to fake the media list
+        guard let mediaList = self.mediaList ?? oneImageMediaList(from: leadImageView.wmf_imageURLToFetch) else {
+            return nil
+        }
+        let previewProvider: UIContextMenuContentPreviewProvider = {
+            let galleryVC = self.getGalleryViewController(for: nil, in: mediaList)
```

Comment:
Should the other gallery previews also use this approach?

Reply:
By "this", do you mean sending the item as `nil`? If so, just updated this to explicitly use the first item in the medialist. It defaults to the first item when initializing `MediaListGalleryViewController`, so this should be functionally identical, yet (helpfully) now more explicit.

Reply:
My bad - I put this comment on the wrong line and wasn't specific enough. I meant the approach of using a single item media list rather than asynchronously requesting the actual media list. If there's no downside to this approach, it might make sense to consolidate the gallery previewing logic through one path.

Reply:
I spun my wheels with that approach for quite a while, got nowhere, and ended up going with this approach. If you (or anyone else) has a suggestion to get the actual media list in this case, I'd love to implement it - I agree it's a much more solid approach. 

What I ran into: `previewProvider` is a `UIContextMenuContentPreviewProvider`, which is a typealias for a function that provides a previewing VC (`() -> UIViewController?`). I don't believe it wants to (or is able to) wait for any sort of async action to get that VC back. On the other hand, our `getMediaList` function asynchronously makes the network request for the media list, and calls the completion argument once the network request returns. I don't believe there is any way to make these two play nice with each other. We could maybe create a new blocking synchronous function to make the network request for the media list, but that seems a bit heavy handed. It's possible there are also other options I'm neglecting to think of.

I think I mentioned this hack in some meeting before settling on it, but you may not have been there. To be clear about the tradeoffs with this approach: When entering gallery view via peek and pop (not via a normal click) AND that is the first image you've tapped on (and thus there is no medialist loaded), you don't get the ability to horizontally scroll between article images. Not the best, but given how corner case it is, it seemed acceptable to everyone present.

Reply:
Makes sense - the other option that occurs to me would be immediately returning `UIViewController` subclass that acts a container. It could show a spinner until it receives another `UIViewController` that it adds as a `child` covering its whole view. I don't know how finicky the presentation of the `UIContextMenuContentPreview` is, so it might not work well. Given this, and that there was already a discussion of the tradeoffs, I'm good calling this resolved.

Reply:
Given how rare of a bug this is, I was initially trying to have as light of a fix as possible (both for code complexity purposes, as well as ensuring a small bug didn't eat up a disproportionate amount of time) - not creating new VCs or significant amounts of new logic. That said, let's make it right for the user - I'm happy to spin this up as a new VC. Will keep working on this.

Reply:
Just added this in. I also had to fake a loading screen (for very slow networks). Using the macOS Network Link Conditioner, this seems to fit the bill - always loads all items for the gallery, and if the network request w/ the medialist doesn't come back quick enough we throw up a quick (user-dismissable) loading screen that looks identical to what is used when loading the actual image. In the background, we're fetching the mediaList and then fetching the first image - but loading both segments now look identical to the user (as if it was one load), as they should.